lib/dialects/abstract/query-generator/composition-group.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const Composition = require('./composition');
+
+/*
+ * Holds multiple compositions or objects cohercible to compositions
+ *
+ * Useful for grouping and manipulating anonymous compositions. It
+ * holds compositions or single composition items.
+ *
+ * Concatenable methods.
+ */
+
+class CompositionGroup {
+  constructor(...args) {
+    this.compositions = args.slice(0);
+  }
+  get length() {
+    return this.compositions.length;
+  }
+  add(...compositions) {
+    Array.prototype.push.apply(this.compositions, compositions);
```

Comment:
I think `this.compisitions = this.compisitions.concat(compositions);` or `this.composititions.push(...compositions)` would be cleaner than this prototype hackery.